From 36d5b05f11facb5553910cebc2557befdde3f6aa Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:12:42 -0400
Subject: [virt] kvm/ppc: book3s/hv - use xics_wake_cpu only when defined

Message-id: <1410765214-16377-27-git-send-email-dgibson@redhat.com>
Patchwork-id: 94721
O-Subject: [PATCH 26/78] KVM: PPC: Book3S HV: use xics_wake_cpu only when defined
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1123145

Signed-off-by: David Gibson <dgibson@redhat.com>

RHEL backport note:

    The second hunk of the upstream patch is missing, because it was
    already incorporated in Don Zickus' backport of e0b7ec0 "KVM: PPC:
    Book3S HV: Align physical and virtual CPU thread numbers".

commit 48eaef0518a565d3852e301c860e1af6a6db5a84
Author: Andreas Schwab <schwab@linux-m68k.org>
Date:   Mon Dec 30 15:36:56 2013 +0100

    KVM: PPC: Book3S HV: use xics_wake_cpu only when defined

    Signed-off-by: Andreas Schwab <schwab@linux-m68k.org>
    CC: stable@vger.kernel.org
    Signed-off-by: Alexander Graf <agraf@suse.de>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
 arch/powerpc/kvm/book3s_hv.c
---

diff --git a/arch/powerpc/kvm/book3s_hv.c b/arch/powerpc/kvm/book3s_hv.c
index b6c31d3..ceefd84 100644
--- a/arch/powerpc/kvm/book3s_hv.c
+++ b/arch/powerpc/kvm/book3s_hv.c
@@ -88,10 +88,13 @@ static void kvmppc_fast_vcpu_kick_hv(struct kvm_vcpu *vcpu)
 
  /* CPU points to the first thread of the core */
  if (cpu != me && cpu >= 0 && cpu < nr_cpu_ids) {
+#ifdef CONFIG_KVM_XICS
   int real_cpu = cpu + vcpu->arch.ptid;
   if (paca[real_cpu].kvm_hstate.xics_phys)
    xics_wake_cpu(real_cpu);
-  else if (cpu_online(cpu))
+  else
+#endif
+  if (cpu_online(cpu))
    smp_send_reschedule(cpu);
  }
  put_cpu();
-- 
1.7.1